### PR TITLE
protected routing, db policies, gallery mood filter

### PR DESCRIPTION
**Pull request by alternadiva**

### CODE REVIEW

File: components/authentication/Login.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import { supabase } from "../../utils/supabaseClient";
+
+function Login() {
+  const [mail, setMail] = useState("");
+  const [pass, setPass] = useState("");
+
+  const router = useRouter();
+
+  const signIn = async () => {
```
Comment (PJSalter) [inline components/authentication/Login.js:13]: really like the use of async for your signIn page, really good work :sparkles:

File: components/authentication/Login.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    const { user, session, error } = await supabase.auth.signIn({
```
Comment (PJSalter) [inline components/authentication/Login.js:14]: super kool auth with supabase on awaiting the user info ✨

File: pages/gallery.js [hunk 3]
```diff
@@ -1,17 +1,57 @@
...
 
   return (
-    <div>
-      <h2>Gallery</h2>
-    </div>
+    <section>
+      <FilterNav moods={moods} arts={allArts} mood={mood} setMood={setMood} />
+      <ul>
+        {allArts
+          .filter((art) => mood === "all" || art.mood === mood)
```
Comment (PJSalter) [inline pages/gallery.js:20]: awesome 🤩

File: pages/gallery.js [hunk 4]
```diff
@@ -1,17 +1,57 @@
-import { useEffect } from "react";
+import { useState } from "react";
+import { useRouter } from "next/router";
 
-import MoodCollection from "../components/filter/MoodCollection";
+import FilterNav from "../components/filter/FilterNav";
```
Comment (PJSalter) [inline pages/gallery.js:4]: great use of creating the filterNav and importing this path 🌟

File: pages/upload.js [hunk 5]
```diff
@@ -26,37 +31,46 @@ function Upload(props) {
...
+      console.log("else statement");
       const { data, error } = await supabase
-        .from("moods")
-        .insert([
-          {
-            mood: inputs.mood,
-            count: 1,
-          },
-        ])
+        .rpc("increment", {
```
Comment (AsmahanM) [inline pages/upload.js:58]: Filtering the moods will definitely improve the user experience. I like how you're listening out to see if the mood exists and keeping count

**Review by AsmahanM — APPROVED** [review]

File: pages/api/auth/index.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import { supabase } from "../../../utils/supabaseClient";
+
+function handler(req, res) {
```
Comment (PJSalter) [inline pages/api/auth/index.js:3]: really kool how you have implemented a handler for setting the auth cookies 🍪

File: pages/_app.js [hunk 7]
```diff
@@ -1,12 +1,69 @@
...
+    useState("not-authenticated");
+
+  const router = useRouter();
+
+  useEffect(() => {
+    /* fires when a user signs in or out */
+    const { data: authListener } = supabase.auth.onAuthStateChange(
+      (event, session) => {
+        handleAuthChange(event, session);
+        if (event === "SIGNED_IN") {
```
Comment (PJSalter) [inline pages/_app.js:24]: ✨ 🤩 very clear and neat to understand

**Review by PJSalter — APPROVED** [review]

**Pull request merged**